### PR TITLE
18684 map config revise map config

### PR DESCRIPTION
**Pull request by robertvilla**
Initial commit to get the current Roam map config and maps to work with the new roam interface. Note: there are various elements to be fixed, e.g. some hardcoded config, etc. 

To see the maps you have to be logged in to dev (for development purposes I logged into dev roam before then running the new roam). 

Redmine issue: https://redmine.edina.ac.uk/issues/18684

### CODE REVIEW

File: proxy/api/mapconfig.js [hunk 1]
```diff
@@ -1,101 +1,18 @@
...
-      attributions: ['Astun Data Service &copy; Ordnance Survey.'],
-      sublayers: ['osopen'],
-      format: 'image/png',
-      opacity: 1,
-    }],
-  }
-}
+let fs = require('fs');
 
 let getMapConfig = (req, res) =>{
```
Comment (marksmall) [inline proxy/api/mapconfig.js]: Very minor, but we don't need to do `=>{` anymore, better with a space e.g. `=> {`. The interface dictated we didn't have a trailing space after the **fat arow**, but this demo is different

File: proxy/api/mapconfig.js [hunk 2]
```diff
@@ -1,101 +1,18 @@
...
   let mapId = req.params.mapId
```
Comment (marksmall) [inline proxy/api/mapconfig.js]: Prefer **const** to **let** if you don't change the value

**Comment by marksmall** [comment]
something that has bugged me is the lack of comments for our  map config. This is because it is JSON and cannot contain comments, but we can easily overcome this by using **YAML** instead. The config files are complicated, elements are re-used around the application, therefore what may be considered a simple change can have far reaching impact. This could be explained in comments, if we could

**Comment by robertvilla** [comment]
It might be good to move the configuration, at least partially, to the DB. It would make it easier for the data team to alter the map data, such as the dates, without us being involved. A problem with this is keeping the different environments up to date or in sync.

How big a problem this could be might be dependent on the data in question: perhaps putting user facing data (names, dates, etc) in the DB would be good, and keep any infrastructure data in properties files which can vary by environment via puppet. 

Yes, comments in the property files would be good. Whether YAML is the best choice I'm not sure at the moment.

**Comment by robertvilla** [comment]
I've created a new issue for the above: 
https://redmine.edina.ac.uk/issues/18800

**Comment by agxs** [comment]
Yup, the map config would be much better in the DB or something like that. I don't think either switching to a different format or storage place is something we should consider at the moment though. Less that 7 weeks to go!

**Comment by marksmall** [comment]
Good comments Robert, I suggested YAML as it is basically JSON, so the effort should be minimal. Andrew is right though, while this is good to do, it is not necessary and I'm afraid time is the biggest factor. Leave it for now, as long as it is detailed in an issue, we can decide at a later date what the right solution is.

**Pull request merged**